### DISCUSSION TITLE
Is it possible to bind a chart to a ref in Vue.js 3.x?

### DISCUSSION THREAD

**Discussion by dangerlettuce** [naver/billboard.js#3847] [Q&A]
I'm trying to create a chart bound to a ref in Vue.js and it keeps adding the chart at the bottom of the screen, which seems to be default behavior when it can't find a div with ID "chart".  I found an example of someone using Vue 2 this.$refs.chart, but that no longer works in vue 3, possibly after this default behavior was added.  Is this no longer possible?  Or what am I doing wrong?

Example code:
`<template>
    <div ref="bbChart"></div>
</template>
<script setup lang="ts">
    import { ref, onMounted } from 'vue'
    import bb, {bar} from 'billboard.js'
    
    const chartOptions = {
        bindto: bbChart.value,
       data: {
           type: bar(),
           columns: [
              ['x', 2012, 2013, 2014, 2015, 2016, 2017],
              ['data1', 30, 200, 100, 170, 150, 250],
           ],
           x: 'x',
       }
     onMounted(() => {
         bb.generate(chartOptions)
     })
`

**Comment by SbstnErhrdt** [comment]
Yes

```vue
<script setup lang="ts">
import { ref, onMounted, onBeforeUnmount } from "vue";
import bb, { bar } from "billboard.js";
import "billboard.js/dist/billboard.css";

const chartContainer = ref<HTMLElement | null>(null);
let chart: bb.Chart | null = null;

onMounted(() => {
    chart = bb.generate({
        bindto: chartContainer.value!,
        data: {
            columns: [
                ["data1", 30, 200, 100, 400, 150, 250],
                ["data2", 130, 100, 140, 200, 150, 50],
            ],
            type: bar(), // ESM-safe
        },
        bar: { width: { ratio: 0.5 } },
    });
});

onBeforeUnmount(() => {
    chart?.destroy(); // tidy-up when the component leaves
});
</script>

<template>
    <div ref="chartContainer" />
</template>

```
